src/vivarium_gates_mncnh/components/lbwsg.py

```diff
@@ -474,6 +473,107 @@ def calculate_mortality_weights(self, sex: str) -> pd.Series:
         return weights
 
 
+class PretermPrevalenceObserver(Component):
+    CONFIGURATION_DEFAULTS = {
+        "stratification": {
+            "preterm_prevalence": {
+                "exclude": [],
+                "include": [],
+            }
+        }
+    }
+
+    @property
+    def columns_required(self) -> list[str] | None:
+        return ["gestational_age_exposure", "lbwsg_category", "child_alive", "sex_of_child"]
+
+    # noinspection PyAttributeOutsideInit
+    def setup(self, builder: Builder) -> None:
+        self.birth_exposure = builder.data.load(data_keys.LBWSG.BIRTH_EXPOSURE)
+        self.lbwsg_categories = builder.data.load(data_keys.LBWSG.CATEGORIES)
+        self.config = builder.configuration.stratification.preterm_prevalence
+        self.pop_size = builder.configuration.population.population_size
+        self.step_number = 1
+
+        builder.results.register_stratified_observation(
+            name=f"calculated_late_neonatal_preterm_prevalence",
+            aggregator=self.calculate_preterm_prevalence,
+            results_updater=self.results_updater,
+            additional_stratifications=self.config.include,
+            excluded_stratifications=self.config.exclude,
+            when="time_step",
+            to_observe=self.to_observe,
+        )
+
+    def on_time_step_cleanup(self, event: Event) -> None:
+        """Increment step number at the end of each time step."""
+        self.step_number += 1
+
+    def calculate_preterm_prevalence(self, x: pd.DataFrame) -> float:
+        # weight preterm prevalence by fraction of simulants who survived to late neonatal period
```

Comment:
This is all the same as the PAF observer until we get to "Get preterm categories".

Reply:
```python

def calculate_preterm_prevalence(self, x):
    helper_func(self, x)


def helper_func(component, x):
    

```